sdlib/d/gc/extent.d

```diff
@@ -253,6 +256,32 @@ public:
 		return _metadata.slabData;
 	}
 
+	/**
+	 * Label features for slabs.
+	 */
+
+	@property
+	bool allowsLabels() const {
+		return isSlab() && isAppendableSizeClass(sizeClass);
+	}
+
+	uint labelsIndex(uint index) {
+		assert(index < maxLabels, "Invalid labels index!");
+		return index + maxLabels;
+	}
+
+	bool hasLabel(uint index) {
+		return allowsLabels && slabData.valueAt(labelsIndex(index));
+	}
+
+	void clearLabel(uint index) {
+		slabData.clearBit(labelsIndex(index));
+	}
+
+	void setLabel(uint index) {
+		slabData.setBit(labelsIndex(index));
+	}
```

Comment:
This needs to be done atomically. While no other thread can touch this specific allocation, another thread might want to play with the one before/after.

Reply:
Because computer don't manipulate bits, but at best bytes.

Reply:
No, it shouldn't be.